profitbricks/resource_profitbricks_server.go

```
@@ -300,8 +305,13 @@ func resourceProfitBricksServerCreate(d *schema.ResourceData, meta interface{})
 			if rawMap["availability_zone"] != nil {
 				availabilityZone = rawMap["availability_zone"].(string)
 			}
-			if image == "" && licenceType == "" {
-				return fmt.Errorf("Either 'image', or 'licenceType' must be set.")
+			if image == "" && licenceType == "" && !IsSnapshot {
+				return fmt.Errorf("Either 'image',  or 'licenceType', must be set.")
+			}
+
+			if IsSnapshot == true {
```

I would return an error if ssh key or password are passed with snapshot id